tests/integration/test_charm.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# Copyright 2024 Ubuntu
+# See LICENSE file for licensing details.
+
+import logging
+import time
+from pathlib import Path
+
+import lightkube
+import pytest
+import yaml
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from lightkube import codecs
+from lightkube.generic_resource import (
+    create_namespaced_resource,
+    load_in_cluster_generic_resources,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CHARM_NAME = METADATA["name"]
+RESOURCE_DISPATCHER_CHARM_NAME = "resource-dispatcher"
+METACONTROLLER_CHARM_NAME = "metacontroller-operator"
+TESTING_LABELS = ["user.kubeflow.org/enabled"]
+PODDEFAULTS_CRD_TEMPLATE = "./tests/integration/crds/poddefaults.yaml"
+PODDEFAULT_FILE = yaml.safe_load(Path("./src/templates/poddefault.yaml").read_text())
+PODDEFAULT_NAME = PODDEFAULT_FILE["metadata"]["name"]
+NAMESPACE_FILE = "./tests/integration/namespace.yaml"
+
+
+PodDefault = create_namespaced_resource("kubeflow.org", "v1alpha1", "PodDefault", "poddefaults")
+
+
+def _safe_load_file_to_text(filename: str) -> str:
+    """Returns the contents of filename if it is an existing file, else it returns filename."""
+    try:
+        text = Path(filename).read_text()
+    except FileNotFoundError:
+        text = filename
+    return text
+
+
+def delete_all_from_yaml(yaml_text: str, lightkube_client: lightkube.Client = None):
+    """Deletes all k8s resources listed in a YAML file via lightkube.
+
+    Args:
+        yaml_file (str or Path): Either a string filename or a string of valid YAML.  Will attempt
+                                 to open a filename at this path, failing back to interpreting the
+                                 string directly as YAML.
+        lightkube_client: Instantiated lightkube client or None
+    """
+
+    if lightkube_client is None:
+        lightkube_client = lightkube.Client()
+
+    for obj in codecs.load_all_yaml(yaml_text):
+        lightkube_client.delete(type(obj), obj.metadata.name)
+
+
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    client = lightkube.Client(field_manager=CHARM_NAME)
+    return client
+
+
+def deploy_k8s_resources(template_files: str):
+    lightkube_client = lightkube.Client(field_manager=CHARM_NAME)
+    k8s_resource_handler = KubernetesResourceHandler(
+        field_manager=CHARM_NAME, template_files=template_files, context={}
+    )
+    load_in_cluster_generic_resources(lightkube_client)
```

Comment:
Does this call actually do anything? Unless you apply CRDs before calling `deploy_k8s_resources()`, you won't get any CRD loaded.